### PR TITLE
golint warnings

### PR DESCRIPTION
**Pull request by dleonard00**
Begins work on [issue 582](https://github.com/ethersphere/go-ethereum/issues/582)

### CODE REVIEW

**Comment by dleonard00** [comment]
I have a few questions about this that I'll be asking tomorrow in standup. But basically, I'm wanting to make sure I'm doing this right before expanding the scope to other files.

File: swarm/services/swap/swap.go [hunk 1]
```diff
@@ -102,12 +105,12 @@ func NewDefaultSwapParams() *SwapParams {
 	}
 }
 
-//this can only finally be set after all config options (file, cmd line, env vars)
-//have been evaluated
-func (self *SwapParams) Init(contract common.Address, prvkey *ecdsa.PrivateKey) {
+// Init this can only finally be set after all config options (file, cmd line, env vars)
+// have been evaluated
+func (localProfile *LocalProfile) Init(contract common.Address, prvkey *ecdsa.PrivateKey) {
```
Comment (nonsense) [inline swarm/services/swap/swap.go]: Thanks for fixing some of the linter problems and improving godoc!

In Go, when the context is small enough, you don't need to be very explicit with variable names - for example `localProfile` could very well just be `lp` in this scenario.

See: https://www.lysator.liu.se/c/pikestyle.html which is predecessor of many of the ideas for Go.

File: swarm/services/swap/swap.go [hunk 2]
```diff
@@ -184,66 +186,69 @@ func NewSwap(local *SwapParams, remote *SwapProfile, backend chequebook.Backend,
 	return
 }
 
-func (self *SwapParams) Chequebook() *chequebook.Chequebook {
-	defer self.lock.Unlock()
-	self.lock.Lock()
-	return self.chbook
+// Chequebook get's chequebook from the localProfile
+func (localProfile *LocalProfile) Chequebook() *chequebook.Chequebook {
```
Comment (nonsense) [inline swarm/services/swap/swap.go]: Same here. Just `lp` as the context is very small.

**Comment by zelig** [comment]
so far looks good to me

File: swarm/services/swap/swap.go (hunk 1 above)
Comment (dleonard00) [reply swarm/services/swap/swap.go]: Thanks for this feedback and resource, happy to make these changes and work to adapt this style into my future commits. 👍

**Comment by dleonard00** [comment]
Go ahead and merge this if test's pass. I will create separate pull requests for the rest of this.

**Review by acud — APPROVED** [review]

File: swarm/services/swap/swap/swap.go [hunk 3]
```diff
@@ -93,160 +92,161 @@ type Swap struct {
...
+// Payment handlers
 type Payment struct {
 	Out         OutPayment // outgoing payment handler
 	In          InPayment  // incoming  payment handler
 	Buys, Sells bool
 }
 
-// swap constructor
-func New(local *Params, pm Payment, proto Protocol) (self *Swap, err error) {
+// New - swap constructor
```
[the remark this answers is not in the document]
Comment (nonsense) [reply swarm/services/swap/swap/swap.go:102]: @justelad I'd say no - it is part of the `swap` package already, so it is clear.

**Pull request merged**